bundles/org.openhab.binding.cbus/src/main/java/org/openhab/binding/cbus/handler/CBusNetworkHandler.java

```diff
@@ -0,0 +1,290 @@
+                if (session != null) {
+                    try {
+                        projectObject = (Project) session.getCGateObject("//" + project);
+                        this.projectObject = projectObject;
+                    } catch (CGateException e2) {
+                        // We dont need to do anything other than stop this propagating
+                    }
+                }
+                if (projectObject == null) {
+                    logger.debug("Cant get projectobject");
+                    return;
+                }
+            }
+            logger.debug("projectobject {}", projectObject);
+            if (network == null) {
+                CGateSession session = cbusCGateHandler.getCGateSession();
+                if (session != null) {
+                    try {
+                        network = (Network) session.getCGateObject("//" + project + "/" + networkID);
+                        this.network = network;
+                    } catch (CGateException e2) {
+                        // We dont need to do anything other than stop this propagating
+                    }
+                }
+                if (network == null) {
+                    logger.debug("cgateOnline: Cant get network");
+                    return;
+                }
+            }
+            String state = network.getState();
+            logger.debug("Network state is {}", state);
+            if ("new".equals(state)) {
+                projectObject.start();
+                logger.debug("Need to wait for it to be synced");
+            } else if ("sync".equals(state)) {
+                logger.debug("Network is syncing so wait for it to be ok");
+            }
+            if (!"ok".equals(state)) {
+                ScheduledFuture<?> initNetwork = this.initNetwork;
+                if (initNetwork == null || initNetwork.isCancelled()) {
+                    this.initNetwork = scheduler.scheduleWithFixedDelay(this::checkNetworkOnline, 30, 30,
+                            TimeUnit.SECONDS);
+                    logger.debug("Schedule a check every minute");
+                } else
+                    logger.debug("initNetwork alreadys started");
+                updateStatus();
+                return;
+            }
+            logger.debug("State should be ok !!!!!!");
+        } catch (CGateException e) {
+            logger.warn("Cannot load C-Bus network {}", networkID, e);
+            updateStatus(ThingStatus.UNINITIALIZED, ThingStatusDetail.COMMUNICATION_ERROR);
+        }
+        updateStatus();
+    }
+
+    private void checkNetworkOnline() {
+        Network network = getNetwork();
+        try {
+            if (network != null && network.isOnline()) {
+                logger.debug("Network is online");
+                ScheduledFuture<?> initNetwork = this.initNetwork;
+                if (initNetwork != null) {
+                    initNetwork.cancel(false);
```

Comment:
should you set `initNetwork` to null here as well?

Reply:
I do it now. It would have coped because the only place that mattered it checked isCanceled as well. But this is cleaner.